src/pagination/pagination.tsx

```diff
@@ -0,0 +1,173 @@
+    // EVENT HANDLERS
+    // =============================================================================
+    const handlePaginationItemOnClick = (pageIndex: number) => {
+        if (onPageChange) {
+            onPageChange(pageIndex);
+        }
+    };
+
+    // =============================================================================
+    // RENDER FUNCTIONS
+    // =============================================================================
+    const paginationItemList = [...Array(totalPages)].map((e, i) => {
+        const pageIndex = i + 1;
+        const startRange = boundaryRange + 2 + siblingRange * 2;
+        const endRange = totalPages - startRange;
+        const totalRange = (boundaryRange + siblingRange) * 2 + 3;
+        const active = activePage === pageIndex;
+        if (totalPages <= totalRange) {
+            return (
+                <PageItem
+                    key={pageIndex}
+                    onClick={() => handlePaginationItemOnClick(pageIndex)}
+                    $selected={active}
+                >
+                    {pageIndex}
+                </PageItem>
+            );
+        }
+
+        const ellipsisStart =
+            activePage + siblingRange > startRange &&
+            pageIndex === boundaryRange + 1;
+        const ellipsisEnd =
+            activePage - siblingRange <= endRange &&
+            pageIndex === totalPages - boundaryRange - 1;
+
+        if (ellipsisStart || ellipsisEnd) {
+            return (
+                <EllipsisItem key={pageIndex} focusHighlight={false}>
+                    <EllipsisHorizontalIcon />
+                </EllipsisItem>
+            );
+        }
+
+        const paginationStart =
+            (pageIndex <= startRange &&
+                activePage + siblingRange <= startRange) ||
+            pageIndex <= boundaryRange;
+        const paginationMiddle =
+            pageIndex === activePage ||
+            (pageIndex <= activePage + siblingRange &&
+                pageIndex >= activePage - siblingRange);
+        const paginationEnd =
+            (pageIndex > endRange && activePage - siblingRange > endRange) ||
+            pageIndex > totalPages - boundaryRange;
+        if (paginationStart || paginationMiddle || paginationEnd) {
+            return (
+                <PageItem
+                    key={pageIndex}
+                    onClick={() => handlePaginationItemOnClick(pageIndex)}
+                    $selected={active}
+                >
+                    {pageIndex}
+                </PageItem>
+            );
+        }
+
+        return null;
+    });
+
+    return (
+        <PaginationWrapper
+            className={className}
+            ref={ref}
+            id={id || "pagination-wrapper"}
+            data-testid={dataTestId || "pagination"}
+        >
+            <PaginationList>
+                <PaginationMenu>
+                    {showFirstAndLastNav && (
+                        <NavigationItem
```

Comment:
`aria-label` should still be applied to the buttons. they’re meant to describe the button functionality to screen readers